portfolio/src/main/webapp/script.js

```diff
@@ -114,19 +114,155 @@ function createCommentElement(server_comment) {
 }
 
 /** Tells the server to fetch and delete all comments. */
 function deleteAllComments() {
-    fetch('/data').then(response => response.text()).then((comments) => {
-      const commentsElement = document.getElementById('comments-container');
-      const commentsObj = JSON.parse(comments);
-      commentsObj.forEach((comment) => {
-        deleteComment(comment);
-      })
-      commentsElement.innerText = '';
-    });
-}
\ No newline at end of file
+  fetch('/data').then(response => response.text()).then((comments) => {
+    const commentsElement = document.getElementById('comments-container');
+    const commentsObj = JSON.parse(comments);
+    commentsObj.forEach((comment) => {
+      deleteComment(comment);
+    })
+    commentsElement.innerText = '';
+  });
+}
+
+/**
+ * Creates a map and adds it to the page.
+ */
+var purpleStyledMap;
+var map;
+var marker;
+function initMap() {
+  // Create a new StyledMapType object, passing it an array of desaturated purple
+  // styles, and the name to be displayed on the map type control. 
+  purpleStyledMap = new google.maps.StyledMapType(
+    [
+      {
+        featureType: 'landscape',
+        stylers: [
+          {
+            "hue": "#84a3f8"
+          },
+          {
+            "saturation": -10
+          }
+        ]
+      },
+      {
+        featureType: 'poi',
+        stylers: [
+          {
+            "hue": "#84a3f8"
+          },
+          {
+            "saturation": -50
+          },
+          {
+            "lightness": -45
+          },
+          {
+            "visibility": "off"
+          }
+        ] 
+      },
+      {
+        featureType: 'road',
+        stylers: [
+          {
+            "hue": "#84a3f8"
+          },
+          {
+            "saturation": -30
+          }
+        ]
+      },
+      {
+        featureType: 'road.local',
+        stylers: [
+          {
+            "lightness": 20
+          }
+        ]
+      },
+      {
+        featureType: 'transit',
+        stylers: [
+          {
+            "hue": "#84a3f8"
+          },
+          {
+            "saturation": -15
+          },
+          {
+            "visibility": "simplified"
+          }
+        ]
+      },
+      {
+        featureType: 'transit.line',
+        stylers: [
+          {
+            "saturation": -70
+          }
+        ]
+      },
+      {
+        featureType: 'water',
+        stylers: [
+          {
+            "hue": "#b7c8f8"
+          },
+          {
+            "saturation": -20
+          },
+          {
+            "lightness": 10
+          }
+        ]
+      }
+    ],
+    {
+      name: 'Purple Map'
+    }
+  );
+
+  map = new google.maps.Map(document.getElementById('map'), {
+    center: {lat: 33.9955566, lng: -118.4768693},
+    zoom: 10,
+    mapTypeControlOptions: {
+     mapTypeIds: ['roadmap', 'satellite', 'terrain', 'styled_map']
+    }
+  });
+
+  // Create a marker.
+  marker = new google.maps.Marker({
```

Comment:
if you are going to create many markers (like those ideas on ur pr comment) perhaps make a method to generate markers with passed in arguments